runtime/lua/vim/lsp/handlers.lua

```diff
@@ -17,6 +18,19 @@ local function err_message(...)
   api.nvim_command("redraw")
 end
 
+--see: https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_semanticTokens
+M['textDocument/semanticTokens/full'] = function(err, result, ctx, config)
+  if not err then
+    semantic_tokens._handle_full(result, ctx, config)
+  end
+end
+
+M['workspace/semanticTokens/refresh'] = function(err, result, ctx, config)
```

Comment:
TODO: how are events handled. This is not a request

Reply:
No, it is a request to the LSP to recompute! It will return error or void. So we can provide an empty handler!